src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberContainerSymbol.cs

```diff
@@ -1370,7 +1377,31 @@ protected MembersAndInitializers GetMembersAndInitializers()
             AddDeclarationDiagnostics(diagnostics);
             diagnostics.Free();
 
-            return membersAndInitializers!;
+            return membersAndInitializers;
+
+            MembersAndInitializersBuilder getDeclaredMembersAndInitializers()
+            {
+                var declaredMembersAndInitializers = _lazyDeclaredMembersAndInitializers;
+                if (declaredMembersAndInitializers != null)
+                {
+                    return declaredMembersAndInitializers;
+                }
+
+                var diagnostics = DiagnosticBag.GetInstance();
+                declaredMembersAndInitializers = BuildDeclaredMembersAndInitializers(diagnostics);
+
+                var alreadyKnown = Interlocked.CompareExchange(ref _lazyDeclaredMembersAndInitializers, declaredMembersAndInitializers, null);
+                if (alreadyKnown is not null)
+                {
+                    diagnostics.Free();
+                    return alreadyKnown;
+                }
+
+                AddDeclarationDiagnostics(diagnostics);
+                diagnostics.Free();
+
+                return declaredMembersAndInitializers!;
```

Comment:
>declaredMembersAndInitializers! [](start = 23, length = 31)

The suppression doesn't look appropriate because it looks like ```BuildDeclaredMembersAndInitializers``` can return null in case of a race. And then we will crash while trying to ```Clone```.